src/main/java/com/minecolonies/core/event/QuestObjectiveEventHandler.java

```diff
@@ -215,4 +228,92 @@ public static void removeKillQuestObjectiveListener(final EntityType<?> entityTo
     {
         entityKillObjectives.getOrDefault(entityToKill, new HashMap<>()).getOrDefault(assignedPlayer, new ArrayList<>()).remove(colonyQuest);
     }
+
+    /**
+     * Research complete event.
+     * @param colony the colony the research happened in.
+     * @param id the research id.
+     */
+    public static void onResearchComplete(final IColony colony, final ResourceLocation id)
+    {
+        if (researchObjectives.containsKey(id))
+        {
+            for (final IQuestInstance instance : new ArrayList<>(researchObjectives.get(id).getOrDefault(colony, new ArrayList<>())))
```

Comment:
do we actually need to copy it into a new list for iteration? Should be fine without since we do not change it during iteration

Reply:
Actually, if it clears the objective, within the call it removes it from the list

Reply:
oh thats fine then